pkg/minishift/registration/registration.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright (C) 2016 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package registration
+
+import (
+	"errors"
+	"github.com/docker/machine/libmachine/log"
+	"github.com/docker/machine/libmachine/drivers"
+)
+
+var (
+	ErrDetectionFailed = errors.New("Distro type not recognized to Registration")
+	registrators       = make(map[string]*RegisteredRegistrator)
+	detector  Detector = &StandardRegistrator{}
+)
+
+
+type SSHCommander interface {
+	// Short-hand for accessing an SSH command from the host.
+	SSHCommand(args string) (string, error)
+}
+
+type Detector interface {
+	DetectRegistrator(d drivers.Driver) (Registrator, error)
+}
+
+type StandardRegistrator struct{}
+
+func SetDetector(newDetector Detector) {
+	detector = newDetector
+}
+
+// Registration defines distribution specific actions
+type Registrator interface {
```

Comment:
As per the Oxford dictionary https://en.oxforddictionaries.com/definition/registrator Registrator is a valid word. (Again that is British English and I think we use American English for docs). So I suppose we can use it, if we wish, though commonly we do use Registrar and most people would identify with it. 
Also Registrar conjures a person whereas Registrator seems impersonal and machine like.
As per https://en.wiktionary.org/wiki/registrator;  'registrator ‎: A registrar or registrant, person or device which registers data.' I find this part 'device which registers data' especially pertinent to this case.